internal/bminventory/inventory.go

```diff
@@ -730,10 +730,13 @@ func (b *bareMetalInventory) integrateWithAMS(ctx context.Context, cluster *comm
 	}
 	log.Infof("AMS subscription %s was created for cluster %s", sub.ID(), *cluster.ID)
 	if err := b.clusterApi.UpdateAmsSubscriptionID(ctx, *cluster.ID, strfmt.UUID(sub.ID())); err != nil {
-		log.WithError(err).Errorf("Failed to update ams_subscription_id in cluster %v, rolling back cluster registration", *cluster.ID)
+		log.WithError(err).Errorf("Failed to update ams_subscription_id in cluster %v, rolling back cluster registration and AMS subscription", *cluster.ID)
 		if deregisterErr := b.clusterApi.DeregisterCluster(ctx, cluster); deregisterErr != nil {
 			log.WithError(deregisterErr).Errorf("Failed to rollback cluster %s registration", *cluster.ID)
 		}
+		if deleteSubErr := b.ocmClient.AccountsMgmt.DeleteSubscription(ctx, strfmt.UUID(sub.ID())); deleteSubErr != nil {
```

Comment:
I think that this should be done before line 734 - DeregisterCluster

Reply:
Changed.